patches/0030-Restore-S3ConnURICleaningDisabled.patch

```diff
@@ -0,0 +1,43 @@
+From 6c2ce6c4b0ea782d1ac45ae32c4592f838fb5f77 Mon Sep 17 00:00:00 2001
+From: Anton Tayanovskyy <anton@pulumi.com>
+Date: Mon, 25 Sep 2023 15:22:30 -0400
+Subject: [PATCH 2/2] Restore S3ConnURICleaningDisabled
+
+---
+ internal/conns/awsclient.go | 9 +++++++++
+ 1 file changed, 9 insertions(+)
+
+diff --git a/internal/conns/awsclient.go b/internal/conns/awsclient.go
+index 40f2242413..7740c0e671 100644
+--- a/internal/conns/awsclient.go
++++ b/internal/conns/awsclient.go
+@@ -10,10 +10,12 @@ import (
+ 	"sync"
+ 
+ 	aws_sdkv2 "github.com/aws/aws-sdk-go-v2/aws"
++	aws_sdkv1 "github.com/aws/aws-sdk-go/aws"
+ 	endpoints_sdkv1 "github.com/aws/aws-sdk-go/aws/endpoints"
+ 	session_sdkv1 "github.com/aws/aws-sdk-go/aws/session"
+ 	apigatewayv2_sdkv1 "github.com/aws/aws-sdk-go/service/apigatewayv2"
+ 	mediaconvert_sdkv1 "github.com/aws/aws-sdk-go/service/mediaconvert"
++	s3_sdkv1 "github.com/aws/aws-sdk-go/service/s3"
+ 	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+ 	"github.com/hashicorp/terraform-provider-aws/names"
+ )
+@@ -61,6 +63,13 @@ func (client *AWSClient) S3UsePathStyle() bool {
+ 	return client.s3UsePathStyle
+ }
+ 
++func (client *AWSClient) S3ConnURICleaningDisabled(ctx context.Context) *s3_sdkv1.S3 {
```

Comment:
Opening a new patch here.

Reply:
https://github.com/pulumi/pulumi-aws/issues/2831 opened a ticket to track the patch.